methcomp/glucose.py

```diff
@@ -807,124 +1251,163 @@ def __init__(self, reference, test, units,
 
     def _check_params(self):
         if len(self.reference) != len(self.test):
-            raise ValueError('Length of reference and test values are not equal')
-
-        if self.units not in ['mmol', 'mg/dl', 'mgdl']:
-            raise ValueError('The provided units should be one of the following: mmol, mgdl or mg/dl.')
-
-        if any([x is not None and not isinstance(x, str) for x in [self.x_title, self.y_title]]):
-            raise ValueError('Axes labels arguments should be provided as a str.')
+            raise ValueError("Length of reference and test values are not equal")
+
+        if self.units not in ["mmol", "mg/dl", "mgdl"]:
+            raise ValueError(
+                "The provided units should be one of the following: "
+                "mmol, mgdl or mg/dl."
+            )
+
+        if any(
+            [
+                x is not None and not isinstance(x, str)
+                for x in [self.x_title, self.y_title]
+            ]
+        ):
+            raise ValueError("Axes labels arguments should be provided as a str.")
 
     def _derive_params(self):
         if self.x_title is None:
-            _unit = 'mmol/L' if 'mmol' else 'mg/dL'
-            self.x_title = 'Reference glucose concentration ({})'.format(_unit)
+            _unit = "mmol/L" if "mmol" else "mg/dL"
+            self.x_title = "Reference glucose concentration ({})".format(_unit)
 
         if self.y_title is None:
-            _unit = 'mmol/L' if 'mmol' else 'mg/dL'
-            self.y_title = 'Predicted glucose concentration ({})'.format(_unit)
-
+            _unit = "mmol/L" if "mmol" else "mg/dL"
+            self.y_title = "Predicted glucose concentration ({})".format(_unit)
 
     def _calc_error_score(self):
-        n = 18 if self.units == 'mmol' else 1
+        n = 18 if self.units == "mmol" else 1
         ref = self.reference * n
         pred = self.test * n
 
         _zones = []
-        from . import static # temporary fix
-        data = np.loadtxt(pkg_resources.open_text(static, 'seg.csv'))
+        from . import static  # temporary fix
+
+        data = np.loadtxt(pkg_resources.open_text(static, "seg.csv"))
         _zones = np.array([data.T[int(p), int(t)] for p, t in zip(pred, ref)])
 
         return _zones
 
     def plot(self, ax):
         # ref, pred
-        ref = self.reference
-        pred = self.test
-        from . import static # temporary fix 
-        _data = np.loadtxt(pkg_resources.open_text(static, 'seg.csv'))
+        #        ref = self.reference NOT USED
```

Comment:
NOTE: flake8 complained about not used, I don't understand what it does, but it is now commented out

Reply:
this is actually correct, the functionality got moved to `.calc_error_score`. If you want, you can remove it for now, but when I will implement the Comparer functionality for the Surveillance Error Grids this is gonna be re-done anyhow.